src/main/java/com/fasterxml/jackson/databind/cfg/JsonNodeFeature.java

```diff
@@ -61,7 +61,25 @@ public enum JsonNodeFeature implements DatatypeFeature
      *
      * @since 2.15
      */
-    STRIP_TRAILING_BIGDECIMAL_ZEROES(true)
+    STRIP_TRAILING_BIGDECIMAL_ZEROES(true),
+
+    /**
+     * Determines the behavior when coercing `NaN` to {@link java.math.BigDecimal} with
+     * {@link com.fasterxml.jackson.databind.DeserializationFeature#USE_BIG_DECIMAL_FOR_FLOATS} enabled.
+     *
+     * 1. If set to {@code true}, will throw an {@link com.fasterxml.jackson.databind.exc.InvalidFormatException} for
+     * attempting to coerce {@code NaN} into {@link java.math.BigDecimal}.
+     * 2. If set to {@code false}, will simply let coercing {@code NaN} into {@link java.math.BigDecimal} happen,
```

Comment:
I think this needs to change: there is no way to coerce `NaN`s -- conversion to some other value could be possible, but we don't do that nor have plans to.

So I think that given the current behavior, let's just say that `false` means that value will be read as `Double` (`DoubleNode`).

I guess this is between two fails, really; fail to read anything (`true`), or fail to create `BigDecimal`-containing node.